### PR TITLE
Feature/os x support

### PR DESCRIPTION
**Pull request by zenangst**
This PR adds support for Imaginary on OS X. It refactors the project structure to separate classes and structs depending on platform.

The inspiration for this PR comes from `SDWebImage`.

### CODE REVIEW

**Comment by vadymmarkov** [comment]
@zenangst ⭐

File: Sources/Shared/Extensions/NSImageView.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    guard let URL = URL else { return }
+
+    let key = URL.absoluteString
+
+    if let fetcher = fetcher {
+      fetcher.cancel()
+      self.fetcher = nil
+    }
+
+    imageCache.object(key) { [weak self] object in
```
Comment (onmyway133) [inline Sources/Shared/Extensions/NSImageView.swift:17]: This piece of code looks the same for `UIImageView` ?

Comment (vadymmarkov) [inline Sources/Shared/Extensions/NSImageView.swift:17]: @onmyway133 https://github.com/hyperoslo/Imaginary/issues/24 😄

Comment (onmyway133) [inline Sources/Shared/Extensions/NSImageView.swift:17]: 😬

**Pull request merged**